Doc/whatsnew/3.9.rst

```diff
@@ -439,6 +439,12 @@ Changes in the Python API
   :data:`~errno.EBADF` error.
   (Contributed by Victor Stinner in :issue:`39239`.)
 
+* :func:`codecs.lookup` now normalizes the encoding name the same way than
```

Comment:
There are other differences. For example, `normalize_encoding("КОИ-8")` returns `"кои_8"`, but `codecs.lookup` normalizes it to `"8"`.

The comment in the sources is also not correct.

Reply:
encodings.normalize_encoding() says "Note that encoding names should be ASCII only." You're correct: "КОИ-8" is normalized to "8" by codecs.lookup() because the C function _Py_normalize_encoding() ignores non-ASCII letters.

I don't know which behavior is correct. It sounds strange to me to have a non-ASCII encoding name. Which encoding is supposed to be used to encoding the encoding name?!? :-D Maybe encodings.normalize_encoding() should also ignore non-ASCII letters, be more strict.

Reply:
I created bpo-39337: codecs.lookup() ignores non-ASCII characters, whereas encodings.normalize_encoding() copies them.

Reply:
> Maybe encodings.normalize_encoding() should also ignore non-ASCII letters, be more strict.

Hm, the annotation of `normalize_encoding` have the words: Note that encoding names should be ASCII only.
+1  for `encodings.normalize_encoding()`  should be similar as `_Py_normalize_encoding()`.
And I created a PR: #22219

Reply:
> There are other differences. For example, normalize_encoding("КОИ-8") returns "кои_8", but codecs.lookup normalizes it to "8".

After #22219 merged, this problem have been fixed(MAYBE `enhanced` will be more exact).